src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -875,8 +877,13 @@ public AbstractStringBuilder append(long l) {
      * @return  a reference to this object.
      */
     public AbstractStringBuilder append(float f) {
-        FloatingDecimal.appendTo(f,this);
+        try {
+            FloatToDecimal.appendTo(f, this);
+        } catch (IOException ignored) {
```

Comment:
What is the motivation for wrapping with IOException?

Reply:
`[Float|Double]ToDecimal` do not have access to `AbstractStringBuilder`, so have to fail over `Appendable`, which can throw `IOException` in `append(*)` methods.

I have to find another way if this wrapping to make the compiler happy is unacceptable.

Reply:
> `[Float|Double]ToDecimal` do not have access to `AbstractStringBuilder`, so have to fail over `Appendable`, which can throw `IOException` in `append(*)` methods.
> 
> I have to find another way if this wrapping to make the compiler happy is unacceptable.

Ah, I haven't used it myself recently enough to recall the details, but I believe the shared secrets mechanism is intended to allow such cross-package access.

If the current code is kept and if the exception can actually be thrown, explicitly throwing an assertion error is preferable to "assert false".